validators/match.go

```diff
@@ -0,0 +1,55 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+const (
+	matchErr = "value must match regex %s"
+)
+
+type matchValidator struct {
+	regex *regexp.Regexp
+}
+
+func Match(regex *regexp.Regexp) tfsdk.AttributeValidator {
+	return matchValidator{
+		regex: regex,
+	}
+}
+
+func (v matchValidator) Description(context.Context) string {
+	return fmt.Sprintf(matchErr, v.regex.String())
```

Comment:
Same comment as before about putting an actual description. Yes there are existing validators that use the error but this was a mistake. I've linked a proper example in another comment.